syft/pkg/cataloger/java/parse_pom_xml.go

```diff
@@ -151,9 +183,15 @@ func resolveProperty(pom gopom.Project, property string) string {
 				part := parts[partNum]
 				for fieldNum := 0; fieldNum < pomValueType.NumField(); fieldNum++ {
 					f := pomValueType.Field(fieldNum)
-					if part == f.Tag.Get("xml") {
+					tag := f.Tag.Get("xml")
```

Comment:
Looks like this was right on the edge for gocognit before the changes - I'll try and pair it down a bit